modules/extensions-support/src/main/java/org/mule/module/extension/internal/config/DeclaredConfiguration.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.module.extension.internal.config;
+
+import org.mule.extension.introspection.ConfigurationModel;
+
+/**
+ * Component to hold a configuration's instance and its name.
```

Comment:
I would say holds/groups configuration's instance related information (ie: not just the instance and the name)